ace-setup-mq/prep-mq.sh

```diff
@@ -2,61 +2,104 @@
+    echo "  -d  <MQ_BIN_PATH>   Required. Specify the mq bin path"
+    echo "  -u  <AUTH_USER>     Required. Specify the user to give authority for mq/ace monitoring"
+    echo "  -c  <CHANNEL_NAME>  Optional. Specify the channel to be created. If not specify, the default channel INSTANA.SENSOR.SVRCONN will be used."  
+    echo "  -m                  Optional. Set up MQ channel for MQ sensor only. If not set, both channel and topic will be created to support both mq sensor and ace sensor."
 }
 
 # 1. Check the current user has authority to execute MQSC commands
 preCheck(){
     # Check user authority
     if groups | tr ' ' '\n' | grep -q '^mqm$'; then
-        echo -e "\033[1;32mINFO: The user launched the script belongs to mqm group\033[0m"
+        echo "INFO: The user launched the script belongs to mqm group"
     else
         echo -e "\033[1;31mERROR: The user launched the script doesn't belong to mqm group, please use an authorized user to execute MQSC commands\033[0m"
         exit 1
     fi
 
     # check mq bin path 
-    echo "Check the MQ_BIN_PATH exists and file setmqenv exists"
+    echo "INFO: Check the MQ_BIN_PATH exists and file setmqenv exists"
     if [ ! -d "$MQ_BIN_PATH" ] || [ ! -f "$MQ_BIN_PATH/setmqenv" ]; then
         echo -e "\033[1;31mERROR: The path $MQ_BIN_PATH or the file $MQ_BIN_PATH/setmqenv does not exist or both don't exist.\033[0m"
         exit 1
     else
-        echo -e "\033[1;32mINFO: The $MQ_BIN_PATH and related file setmqenv exist.\033[0m"
+        echo "INFO: The $MQ_BIN_PATH and related file setmqenv exist."
     fi
 
     # Setup mq environment to accept mqsc command. 
-    echo -e "\033[1;33mINFO: Setup mq environment to accept mqsc command\033[0m"
+    echo "INFO: Setup mq command environment to accept mqsc command"
     . $MQ_BIN_PATH/setmqenv -s 
     if [ $? -eq 0 ]; then
-       echo -e "\033[1;32mINFO: The environment has been set successfully. \033[0m"
+       echo "INFO: The mq command environment has been set successfully."
     else
        echo -e "\033[1;31mERROR: Failed to set the environment\033[0m"
        exit 1
     fi
+
+    # check the nc command for linux and telnet for aix
+    os_type=$(uname -s)
+    if [[ "$os_type" == "Linux" ]]; then 
+        if ! command -v nc >/dev/null 2>&1; then
+            echo -e "\033[1;31mERROR: nc command not found on Linux system, please relaunch the script after you install nc\033[0m"
+            exit 1 
+        else 
+            echo "INFO: nc command is found"
+        fi
+    else
+        if ! command -v telnet >/dev/null 2>&1; then
+            echo -e "\033[1;31mERROR: telnet command not found on non-Linux system, please relaunch the script after you install telnet\033[0m"
+            return 1
+        else 
+            echo "INFO: telnet command is found"
+        fi
+    fi
 }
+
+# check permission
+check_permission() {
+  local permission=$1
+  permissions=$(dspmqaut -m $QMGR_NAME -t qmgr -p $AUTH_USER)
```

Comment:
should we use <MQ_BIN_PATH>/dspmqaut here? In case there are multiple mq installations.

Reply:
I have set up the mq environment with `.  <MQ_BIN_PATH>/setmqenv -s`, so  I think all the following commands should be executed based on this environment.   I have tested on some environments, if not setting it, the dspmqaut command can't be executed successfully.